lightning/src/routing/mod.rs

```diff
@@ -9,5 +9,25 @@
 
 //! Structs and impls for receiving messages about the network and storing the topology live here.
 
-pub mod router;
 pub mod network_graph;
+pub mod router;
+pub mod scorer;
+
+use prelude::*;
+
+use routing::router::RouteHop;
+
+/// An interface used to score payment channels for path finding.
+///
+///	Scoring is in terms of fees willing to be paid in order to avoid routing through a channel.
+///	Allows updating such penalties as payments routed over a particular path succeed or fail.
+pub trait Score {
+	/// Returns the fee willing to be paid to avoid routing through the given channel.
+	fn channel_penalty_msat(&self, short_channel_id: u64) -> u64;
+
+	/// Handles updating channel penalties after failing to route through a channel.
+	fn payment_path_failed(&self, path: &Vec<RouteHop>, short_channel_id: u64);
```

Comment:
Can we drop these methods for now while we figure out what the "right" API is for feeding this data in (they're unused in this PR, I believe?).